tsdb/db_test.go

```diff
@@ -2359,3 +2359,85 @@ func TestDBReadOnlyClosing(t *testing.T) {
 	_, err = db.Querier(0, 1)
 	testutil.Equals(t, err, ErrClosed)
 }
+
+func TestDBReadOnly_Snapshot(t *testing.T) {
+	var (
+		dbBlocks []*BlockMeta
+		dbDir    string
+		logger   = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+		err      error
+		maxt     int64
+	)
+
+	// Boostrap the db.
+	{
+		dbDir, err = ioutil.TempDir("", "test")
+		testutil.Ok(t, err)
+
+		defer func() {
+			testutil.Ok(t, os.RemoveAll(dbDir))
+		}()
+
+		dbBlocks = []*BlockMeta{
+			{MinTime: 1, MaxTime: 1000},
+			{MinTime: 1000, MaxTime: 3001},
+		}
+
+		for _, m := range dbBlocks {
+			createBlock(t, dbDir, genSeriesWithValue(1, 1, m.MinTime, m.MaxTime, func() float64 { return 1 }))
+		}
+
+		// Append data to the WAL.
+		db, err := Open(dbDir, logger, nil, nil)
+		testutil.Ok(t, err)
+		db.DisableCompactions()
+		app := db.Appender()
+		mint := int64(dbBlocks[len(dbBlocks)-1].MaxTime)
+		maxt = mint + 1000
+		for i := 0; i < 1000; i++ {
+			_, err := app.Add(labels.FromStrings(defaultLabelName, "0"), mint+int64(i), 1.0)
+			testutil.Ok(t, err)
+		}
+		testutil.Ok(t, app.Commit())
+		testutil.Ok(t, app.Rollback())
```

Comment:
Why the rollback call here?

Reply:
ah, not sure; I saw it on the other tests and thought it was boilerplate. I'll remove it and rebase the PR :+1: